backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/emu/com/badlogic/gdx/graphics/glutils/IndexBufferObject.java

```diff
@@ -134,17 +134,20 @@ public void updateIndices (int targetOffset, short[] indices, int offset, int co
 		}
 	}
 
-	/** <p>
-	 * Returns the underlying ShortBuffer. If you modify the buffer contents they wil be uploaded on the call to {@link #bind()}.
-	 * If you need immediate uploading use {@link #setIndices(short[], int, int)}.
-	 * </p>
-	 * 
-	 * @return the underlying short buffer. */
+	/** @deprecated use {@link #getBuffer(boolean)} instead */
+	@Override
+	@Deprecated
 	public ShortBuffer getBuffer () {
 		isDirty = true;
 		return buffer;
 	}
 
+	@Override
+	public ShortBuffer getBuffer (boolean forWriting) {

```

Comment:
Shouldn't a getter be read-only by default, as the name implies (GET)
marking a buffer as dirty should be done either via a setter `setBuffer()` or just `markDirty()`

Reply:
It still a getter: original buffer reference is not changed. Basically any getter returning an object doesn't prevent caller to modify its state.
IMO, "forWriting" parameter is simpler to understand and more obvious for the caller to have it at one place.